### PR TITLE
List-n/Archive-n bug fixes, documentation updates, clean code and updated test cases

### PR DESCRIPTION
**Pull request by R-Ramana**
Addressed the following:

- Bug Fixes for list-n, archive-n and unarchive-n
- Test cases for Archive and Unarchive commands
- Cleaned up code (included isArchived in the note class), used formatter in find-n
- Updated documentation (PERT Chart included, Dev/User Guides)

### CODE REVIEW

File: src/main/java/seedu/duke/data/notebook/Note.java [hunk 1]
```diff
@@ -10,7 +10,6 @@
     private String title;
     private ArrayList<String> content;
     private Boolean isPinned;
-    private boolean isArchived;
```
Comment (Chongjx) [inline src/main/java/seedu/duke/data/notebook/Note.java:13]: Why did you remove this variable? I thought it is needed to differentiate the notes when saving to txt file? Or is the archived note list saved in a separate txt file?

**Review by Chongjx — APPROVED** [review]
Looks good to me, just that for the find command, can create a find function in notebook, and move the stream in there.

File: src/main/java/seedu/duke/data/notebook/Note.java (hunk 1 above)
Comment (R-Ramana) [reply src/main/java/seedu/duke/data/notebook/Note.java:13]: No need to check for is archived in my implementation. The note will either be in the archived notebook or in the main notebook. When archiving/unarchiving, the note will be removed and added to the respective notebooks.

Comment (Chongjx) [reply src/main/java/seedu/duke/data/notebook/Note.java:13]: Ya, I am referring to loading from text file, what differentiates them to be loaded into respective array list?

Comment (R-Ramana) [reply src/main/java/seedu/duke/data/notebook/Note.java:13]: How is the files being loaded currently? It traverses notes (from notebook) right? Can just make it traverse both notes and archivedNotes?

Comment (Chongjx) [reply src/main/java/seedu/duke/data/notebook/Note.java:13]: Take a look at her PR

Comment (R-Ramana) [reply src/main/java/seedu/duke/data/notebook/Note.java:13]: Will add it back in that case

File: src/main/java/seedu/duke/data/notebook/Note.java [hunk 2]
```diff
@@ -20,11 +20,10 @@
      * @param content of the note.
      * @param isPinned status of the note.
      */
-    public Note(String title, ArrayList<String> content, Boolean isPinned, boolean isArchived) {
+    public Note(String title, ArrayList<String> content, Boolean isPinned) {
         this.title = title;
         this.content = content;
         this.isPinned = isPinned;
-        this.isArchived = isArchived;
```
Comment (Chongjx) [inline src/main/java/seedu/duke/data/notebook/Note.java:27]: Is the boolean added back in? Looks like it's still removed.

**Review by Chongjx — COMMENTED** [review]
Is the boolean for isArchived readded in?

File: src/main/java/seedu/duke/data/notebook/Note.java (hunk 2 above)
Comment (R-Ramana) [reply src/main/java/seedu/duke/data/notebook/Note.java]: Updated, request for a final review prior to merging

File: src/test/java/seedu/duke/command/FindCommandTest.java [hunk 3]
```diff
@@ -15,46 +17,48 @@
...
+        String expected = "=".repeat(maxRowLength)
+                + FormatterStub.encloseRow(FindCommandStub.execute(keyword))
+                + "=".repeat(maxRowLength) + Formatter.LS
+                + "[96m1. Title: TestNote1 [0m"
+                + Formatter.LS + "default..." + Formatter.LS
+                + "=".repeat(maxRowLength) + Formatter.LS
+                + "[96m2. Title: TestNote2 [0m"
+                + Formatter.LS + "default..." + Formatter.LS
+                + "=".repeat(maxRowLength) + Formatter.LS
+                + "=".repeat(maxRowLength) + Formatter.LS;
```
Comment (Chongjx) [inline src/test/java/seedu/duke/command/FindCommandTest.java:61]: You may want to have an arraylist of note for the expected output and then use the Formatter.formatNotes. In the event that the format changes, you do not have to manually change the expected string again.

File: src/test/java/seedu/duke/command/ArchiveNoteCommandTest.java [hunk 4]
```diff
@@ -0,0 +1,134 @@
...
+
+    @Test
+    void execute_validIndex_returnsArchiveMessage() {
+        int index = 1;
+        String title = NotebookStub.getArchiveNoteTitle(index);
+
+        String expected = "=".repeat(maxRowLength)
+                + FormatterStub.encloseRow(title)
+                + "=".repeat(maxRowLength)
+                + System.lineSeparator();
```
Comment (Chongjx) [inline src/test/java/seedu/duke/command/ArchiveNoteCommandTest.java:63]: All expected string can use the Formatter functions, for this case encloseString(title). Since the purpose of the command test are to compare the result and not the formatting, it is ok to just use those functions.

File: src/test/java/seedu/duke/command/ArchiveNoteCommandTest.java [hunk 5]
```diff
@@ -0,0 +1,134 @@
...
+
+    @Test
+    void execute_invalidIndex_returnsInvalidIndexMessage() {
+        int index = 50;
+        String title = NotebookStub.getArchiveNoteTitle(index);
+
+        String expected = "=".repeat(maxRowLength)
+                + FormatterStub.encloseRow(title)
+                + "=".repeat(maxRowLength)
+                + System.lineSeparator();
```
Comment (Chongjx) [inline src/test/java/seedu/duke/command/ArchiveNoteCommandTest.java:77]: Likewise

**Review by Chongjx — APPROVED** [review]
Looks good overall. Just a suggestion for all command JUnit test can get the expected string through formatter functions instead of manually formatting. The purpose of the test cases here is on the content and not the format.
